crates/astria-sequencer/src/bridge/bridge_lock_action.rs

```diff
@@ -226,18 +246,118 @@ mod tests {
 
         // enough balance; should pass
         let expected_deposit_fee = transfer_fee
-            + get_deposit_byte_len(&Deposit {
+            + calculate_base_deposit_fee(&Deposit {
                 bridge_address,
                 rollup_id,
                 amount: 100,
                 asset: asset.clone(),
                 destination_chain_address: "someaddress".to_string(),
                 source_transaction_id: transaction_id,
                 source_action_index: 0,
-            }) * 2;
+            })
+            .unwrap()
+                * 2;
         state
             .put_account_balance(from_address, &asset, 100 + expected_deposit_fee)
             .unwrap();
         bridge_lock.check_and_execute(&mut state).await.unwrap();
     }
+
+    #[test]
+    fn test_calculate_base_deposit_fee() {
+        // Test for length equality with drastically different int values
+        let deposit = Deposit {
+            bridge_address: astria_address(&[1; 20]),
+            rollup_id: RollupId::from_unhashed_bytes(b"test_rollup_id"),
+            amount: u128::MAX,
+            asset: test_asset(),
+            destination_chain_address: "someaddress".to_string(),
+            source_transaction_id: TransactionId::new([0; 32]),
+            source_action_index: u64::MAX,
+        };
+        let calculated_len_0 = calculate_base_deposit_fee(&deposit).unwrap();
+
+        let deposit = Deposit {
+            bridge_address: astria_address(&[1; 20]),
+            rollup_id: RollupId::from_unhashed_bytes(b"test_rollup_id"),
+            amount: 0,
+            asset: test_asset(),
+            destination_chain_address: "someaddress".to_string(),
+            source_transaction_id: TransactionId::new([0; 32]),
+            source_action_index: 0,
+        };
+        let calculated_len_1 = calculate_base_deposit_fee(&deposit).unwrap();
+        assert_eq!(calculated_len_0, calculated_len_1);
+
+        // Ensure longer asset name results in longer byte length.
+        let deposit = Deposit {
+            bridge_address: astria_address(&[1; 20]),
+            rollup_id: RollupId::from_unhashed_bytes(b"test_rollup_id"),
+            amount: 0,
+            asset: "test_asset".parse().unwrap(),
+            destination_chain_address: "someaddress".to_string(),
+            source_transaction_id: TransactionId::new([0; 32]),
+            source_action_index: 0,
+        };
+        let calculated_len_2 = calculate_base_deposit_fee(&deposit).unwrap();
+        assert!(calculated_len_2 >= calculated_len_1);
```

Comment:
Try to avoid `assert!`. If this fails the reader will not know what exactly failed. It's better to use `assert_eq!`. IMO that's also fine here because all of the values are deterministic.